src/OutputReaders/set_field_time_series.jl

```diff
@@ -7,23 +7,34 @@ using Oceananigans.Architectures: cpu_architecture
 
 iterations_from_file(file) = parse.(Int, keys(file["timeseries/t"]))
 
-find_time_index(time::Number, file_times)       = findfirst(t -> t ≈ time, file_times)
-find_time_index(time::AbstractTime, file_times) = findfirst(t -> t == time, file_times)
+function find_time_index(time::Number, file_times, dt)
+    ϵ = 100 * eps(eltype(file_times))
+    return findfirst(t -> isapprox(t, time, atol=ϵ*dt), file_times)
+end
```

Comment:
what's the `dt` here? I'm bit confused..
is it the "characteristic time scale" mentioned in the first comment of the PR?

I see below that the dt is the average difference in times...

Reply:
oh actually I just noticed this followed from issue #4100! let me read there, things might become clear to me!